components/published-toggle/index.jsx

```diff
@@ -16,6 +17,16 @@ const PublishedToggle = ({
     else toggletext = 'UnPublished'
   }
 
+  const handleChange = useCallback(async event => {
+    const isVisible = event.target.checked
+    try {
+      await onVisibilityChanged(isVisible)
```

Comment:
I think you should not do anything here, just call passed method.

Keep components being pure where it's possible.

Reply:
it has to be done like this. Otherwise, if a request fails user may see the wrong data in table.

Reply:
Anyway we will have better interface once we implement your design. So I would pay much attention to this...

Reply:
The new design may no happen anytime soon just because we do not have enough backend support for that. Also it requires finishing #94.

Is this comment affected by #179?

Reply:
>  we do not have enough backend support for that.

How so? We just need to rewrite how expanded row is shown.

Reply:
If you check the [fresh design](https://www.figma.com/proto/rWs351bJa8uVOSS97Nk8Tm/Content?node-id=78%3A41957&viewport=-256%2C347%2C0.9490244388580322&scaling=min-zoom), there are a few bits that require backend support:
- having document issues right in the document entity because it's show contextually in the document metadata and right in the table
- having DOI source right in the document metadata

Merging everything into a single index takes a lot of effort on the backend.

Reply:
So I rebased the master. Can we proceed with the merge? I will keep is as is As I already said it's [not that easy to implement](https://github.com/oacore/dashboard/pull/179#discussion_r391007921).